packages/docs/components/GreenscreenExamples/index.tsx

```diff
@@ -1,187 +1,150 @@
+
+	const onVideoFrame = useCallback(
+		(frame: CanvasImageSource) => {
+			if (!canvas.current) {
+				return;
+			}
+
+			const context = canvas.current.getContext('2d');
+
+			if (!context) {
+				return;
+			}
+
+			context.drawImage(frame, 0, 0, width, height);
+			const imageFrame = context.getImageData(0, 0, width, height);
+			const {length} = imageFrame.data;
+
+			for (let i = 0; i < length; i += 4) {
+				const red = imageFrame.data[i + 0];
+				const green = imageFrame.data[i + 1];
+				const blue = imageFrame.data[i + 2];
+				if (green > 100 && red < 100 && blue < 100) {
+					imageFrame.data[i + 3] = opacity * 255;
+				}
+			}
+
+			context.putImageData(imageFrame, 0, 0);
+		},
+		[height, opacity, width],
+	);
+
+	return (
+		<AbsoluteFill>
+			<AbsoluteFill>
+				<OffthreadVideo
+					style={{opacity: 0}}
+					onVideoFrame={onVideoFrame}
+					src="https://remotion-assets.s3.eu-central-1.amazonaws.com/just-do-it-short.mp4"
+				/>
+			</AbsoluteFill>
+			<AbsoluteFill>
+				<canvas ref={canvas} width={width} height={height} />
+			</AbsoluteFill>
+		</AbsoluteFill>
+	);
 };
 
 export const VideoOnCanvas: React.FC = () => {
-  const video = useRef<HTMLVideoElement>(null);
-  const canvas = useRef<HTMLCanvasElement>(null);
-  const { width, height } = useVideoConfig();
-
-  const onVideoFrame = useCallback(() => {
-    if (
-      !canvas.current ||
-      !video.current ||
-      !video.current.requestVideoFrameCallback
-    ) {
-      return;
-    }
-
-    const context = canvas.current.getContext("2d");
-
-    if (!context) {
-      return;
-    }
-
-    context.filter = "grayscale(100%)";
-    context.drawImage(video.current, 0, 0, width, height);
-
-    video.current.requestVideoFrameCallback(() => onVideoFrame());
-  }, [height, width]);
-
-  useEffect(() => {
-    if (!video.current?.requestVideoFrameCallback) {
-      return;
-    }
-
-    video.current.requestVideoFrameCallback(() => onVideoFrame());
-  }, [onVideoFrame]);
-
-  return (
-    <AbsoluteFill>
-      <AbsoluteFill>
-        <Video
-          ref={video}
-          style={{ opacity: 0 }}
-          startFrom={300}
-          src="https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4"
-        />
-      </AbsoluteFill>
-      <AbsoluteFill>
-        <canvas ref={canvas} width={width} height={height} />
-      </AbsoluteFill>
-    </AbsoluteFill>
-  );
+	const canvas = useRef<HTMLCanvasElement>(null);
+	const {width, height} = useVideoConfig();
+
+	const onVideoFrame = useCallback(
+		(frame: CanvasImageSource) => {
+			if (!canvas.current) {
+				return;
+			}
+
+			const context = canvas.current.getContext('2d');
+
+			if (!context) {
+				return;
+			}
+
+			context.filter = 'grayscale(100%)';
+			context.drawImage(frame, 0, 0, width, height);
+		},
+		[height, width],
+	);
+
+	return (
+		<AbsoluteFill>
+			<AbsoluteFill>
+				<OffthreadVideo
+					onVideoFrame={onVideoFrame}
+					style={{opacity: 0}}
+					startFrom={300}
```

Comment:
This works nicely if startFrom > 0. Consider a situation when startFrom=0 and there are multiple sequences and generating  video using Lambda. 
if startFrom=0, generally canvas sync will show blankscreen, since the video is not started. To remediate we have to offset 0th second to 0.1 frame to show the initial start screen to sync with canvas and avoid having the empty screen.

Reply:
@khanzzirfan It has nothing to do with startFrom, it works with any value.

I only put 300 as value because the video starts with a black frame.